optimizer/resolver.go

```diff
@@ -126,6 +126,13 @@ func (nr *nameResolver) Enter(inNode ast.Node) (outNode ast.Node, skipChildren b
 			// the resolving rule is different from order by clause.
 			nr.currentContext().inByItemExpression = true
 		}
+		if nr.currentContext().inGroupBy {
+			// make sure item is not aggregate function
+			if ast.HasAggFlag(v.Expr) {
+				nr.Err = errors.New("group by cannot contain aggregate function")
```

Comment:
We should convert ColumnNameExpr to AggregateFuncExpr. We may meet ColunmNameExpr before we meet groupby.